crypto-ffi/bindings/jvm/src/main/kotlin/com/wire/crypto/CoreCrypto.kt

```diff
@@ -7,177 +7,31 @@ import kotlinx.coroutines.NonCancellable
-    com.wire.crypto.uniffi.CoreCryptoLogLevel.WARN -> CoreCryptoLogLevel.WARN
-    com.wire.crypto.uniffi.CoreCryptoLogLevel.ERROR -> CoreCryptoLogLevel.ERROR
-}
-
-/** The logger interface */
-interface CoreCryptoLogger {
-    /**
-     *  Core Crypto will call this method whenever it needs to log a message.
-     */
-    fun log(level: CoreCryptoLogLevel, message: String, context: String?)
-}
-
-/**
- * Initializes the logging inside Core Crypto. Not required to be called and by default there will be no logging.
+ * Instantiate a history client.
  *
- * @param logger a callback to implement the platform specific logging. It will receive the string with the log text from Core Crypto
- **/
-fun setLogger(logger: CoreCryptoLogger) {
-    com.wire.crypto.uniffi.setLoggerOnly(object : com.wire.crypto.uniffi.CoreCryptoLogger {
-        override fun log(level: com.wire.crypto.uniffi.CoreCryptoLogLevel, message: String, context: String?) {
-            logger.log(level.lift(), message, context)
-        }
-    })
-}
+ * This client exposes the full interface of `CoreCrypto`, but it should only be used to decrypt messages.
+ * Other use is a logic error.
+ */
+suspend fun historyClient(historySecret: HistorySecret) = coreCryptoHistoryClient(historySecret).lift()
 
 /**
- * Set maximum log level of logs which are forwarded to the [CoreCryptoLogger].
+ * A high-level wrapper around a CoreCrypto client as emitted by Uniffi.
  *
- * @param level the max level that should be logged, by default it will be WARN
+ * This wrapper should be largely transparent to end users. It exists to improve the
+ * callback interfaces: `.transaction(...)`, `.registerFooObserver(...)`, etc.
  */
-fun setMaxLogLevel(level: CoreCryptoLogLevel) {
-    com.wire.crypto.uniffi.setMaxLogLevel(level.lower())
-}
-
-/** The type representing a CoreCrypto client */
-class CoreCrypto(private val cc: com.wire.crypto.uniffi.CoreCrypto) {
-    companion object {
-        internal const val DEFAULT_NB_KEY_PACKAGE: UInt = 100U
-
-        /** Opens an existing core crypto client or creates a new one if one doesn't exist at the `keystore` path */
-        suspend operator fun invoke(
-            keystore: String,
-            databaseKey: DatabaseKey
-        ): CoreCrypto {
-            val cc = com.wire.crypto.uniffi.coreCryptoDeferredInit(keystore, databaseKey.bytes, null)
-            return CoreCrypto(cc)
-        }
-
-        /**
-         * Instantiate a history client.
-         *
-         * This client exposes the full interface of `CoreCrypto`, but it should only be used to decrypt messages.
-         * Other use is a logic error.
-         */
-        suspend fun historyClient(historySecret: HistorySecret): CoreCrypto {
-            val cc = com.wire.crypto.uniffi.coreCryptoHistoryClient(historySecret.lower())
-            return CoreCrypto(cc)
-        }
-    }
-
-    internal fun lower() = cc
+class CoreCryptoHigh(private val cc: CoreCrypto) {
```

Comment:
So the class clients interact is now called `CoreCryptoHigh`. Didn't we discuss renaming the uniffi class to `_CoreCrypto` or `CoreCryptoUniffi` and the class expose publicly will stay as `CoreCrypto`

Reply:
Clients should not need to ever write down `CoreCryptoHigh`; we expose relevant methods on `CoreCrypto` that look like constructors but return a `CoreCryptoHigh` instance:

https://github.com/wireapp/core-crypto/blob/52abdd88222666ede988f8575f58e8d74f036d3d/crypto-ffi/bindings/jvm/src/main/kotlin/com/wire/crypto/CoreCrypto.kt#L13-L25